### PR TITLE
Refactor: Update `CompactCount` to use `PluralCategory` enum

### PR DESCRIPTION
**Pull request by younies**
Related to issue https://github.com/unicode-org/icu4x/issues/5369.

### CODE REVIEW

**Review by robertbastian — DISMISSED** [review]

**Review by robertbastian — APPROVED** [review]

File: provider/source/src/currency/compact.rs [hunk 1]
```diff
@@ -64,12 +64,12 @@ impl DataProvider<ShortCurrencyCompactV1Marker> for SourceDataProvider {
             };
 
             let count = match count_str {
-                "zero" => Count::Zero,
-                "one" => Count::One,
-                "two" => Count::Two,
-                "few" => Count::Few,
-                "many" => Count::Many,
-                "other" => Count::Other,
+                "zero" => PluralCategory::Zero,
```
Comment (robertbastian) [inline provider/source/src/currency/compact.rs:67]: You can use `PluralCategory::get_for_cldr_bytes` here

File: components/experimental/src/dimension/provider/compact_count_ule.rs [hunk 2]
```diff
@@ -65,12 +65,12 @@ impl AsULE for CompactCount {
     #[inline]
     fn from_unaligned(unaligned: Self::ULE) -> Self {
         let count = match unaligned.0 & 0b0000_0111 {
-            0 => Count::Zero,
-            1 => Count::One,
-            2 => Count::Two,
-            3 => Count::Few,
-            4 => Count::Many,
-            5 => Count::Other,
+            0 => PluralCategory::Zero,
```
Comment (robertbastian) [inline components/experimental/src/dimension/provider/compact_count_ule.rs:68]: You can use `(unaligned.0 & 0b0000_0111) as PluralCategory`, as it's `repr(u8)` with stable discriminants

Comment (younies) [reply components/experimental/src/dimension/provider/compact_count_ule.rs:68]:
```
error[E0605]: non-primitive cast: `u8` as `PluralCategory`
  --> components/experimental/src/dimension/provider/compact_count_ule.rs:67:22
   |
67 |         let count =  (unaligned.0 & 0b0000_0111) as PluralCategory;
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object


```

Comment (younies) [reply components/experimental/src/dimension/provider/compact_count_ule.rs:68]: https://github.com/unicode-org/icu4x/pull/5406

**Pull request merged**